File: backend/src/main/java/com/efedotov/vaultly/dto/folder/FolderCreateDto.java
package com.efedotov.vaultly.dto.folder;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.UUID;

@Data
public class FolderCreateDto {
    @NotBlank(message = "Имя папки обязательно")
    @Size(max = 255)
    private String name;

    private UUID parentFolderId;

    private Boolean isHidden = false;

    @Size(max = 255)
    private String hiddenFolderKey;

    @Size(max = 128)
    private String password;

    @Size(max = 500)
    private String description;
}

File: lib/server/model/folder/folder_create_dto.dart
class FolderCreateDto {
  final String name;
  final String? parentFolderId;
  final bool isHidden;
  final String? hiddenFolderKey;
  final String? password;
  final String? description;

  FolderCreateDto({
    required this.name,
    this.parentFolderId,
    this.isHidden = false,
    this.hiddenFolderKey,
    this.password,
    this.description,
  });

  factory FolderCreateDto.fromJson(Map<String, dynamic> json) {
    return FolderCreateDto(
      name: json['name'] as String,
      parentFolderId: json['parentFolderId'] as String?,
      isHidden: json['isHidden'] as bool? ?? false,
      hiddenFolderKey: json['hiddenFolderKey'] as String?,
      password: json['password'] as String?,
      description: json['description'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
    'name': name,
    if (parentFolderId != null) 'parentFolderId': parentFolderId,
    'isHidden': isHidden,
    if (hiddenFolderKey != null) 'hiddenFolderKey': hiddenFolderKey,
    if (password != null) 'password': password,
    if (description != null) 'description': description,
  };
}
